File: LeadToOffer/41_ContinuousSum.cpp
// Contiue sum using two pointer. O(n) time, O(1) space.
class Solution {
public:
    vector<vector<int> > FindContinuousSequence(int sum) {
        vector<vector<int> > res;
        if (sum<3)
            return res;
        int low = 1;
        int high = 2;
        int sum_seq = 0;
        while(low<(sum+1)/2) {
            if (low==high)
                high++;
            sum_seq = (low+high)*(high-low+1)/2;
            if (sum_seq==sum) {
                vector<int> seq;
                for (int i=low; i<=high; i++)
                    seq.push_back(i);
                res.push_back(seq);
                high++;
            } else if (sum_seq<sum) {
                high++;
            } else {
                low++;
            }
        }
        return res;
    }
};

File: Algo/LeadToOffer/41_ContinuousSum.py
class Solution(object):
    def findContinuousSequence(self, target):
        """
        :type target: int
        :rtype: List[List[int]]
        """
        i, j = 1, 2
        res = []
        while i < (target + 1) / 2:
            if i == j:
                j += 1

            if calSum(i, j) == target:
                res.append(list(range(i, j+1)))
                j += 1
            elif calSum(i, j) < target:
                j += 1
            else:
                i += 1

        return res

def calSum(i, j):
    return (i + j) * (j - i + 1) / 2